cnf-certification-test/performance/suite.go

```diff
@@ -101,51 +101,39 @@ func CheckProbePeriodSeconds(elem *v1.Probe, cut *provider.Container, s string)
 	var compliantObjects []*testhelper.ReportObject
 	var nonCompliantObjects []*testhelper.ReportObject
 	counter := 0
 	for _, put := range env.Pods {
 		for _, cut := range put.Containers {
-			if cut.LivenessProbe != nil && cut.LivenessProbe.Exec != nil {
-				counter++
-				if CheckProbePeriodSeconds(cut.LivenessProbe, cut, "LivenessProbe") {
-					compliantObjects = append(compliantObjects, testhelper.NewContainerReportObject(put.Namespace, put.Name,
-						cut.Name, fmt.Sprintf("LivenessProbe exec probe has a PeriodSeconds greater than 10 ( %d seconds)",
-							cut.LivenessProbe.PeriodSeconds), true))
-				} else {
-					nonCompliantObjects = append(nonCompliantObjects,
-						testhelper.NewContainerReportObject(put.Namespace, put.Name,
-							cut.Name, fmt.Sprintf("LivenessProbe exec probe has a PeriodSeconds that is not greater than 10 ( %d seconds)",
-								cut.LivenessProbe.PeriodSeconds), false))
-				}
-			}
-			if cut.StartupProbe != nil && cut.StartupProbe.Exec != nil {
-				counter++
-				if CheckProbePeriodSeconds(cut.StartupProbe, cut, "StartupProbe") {
-					compliantObjects = append(compliantObjects, testhelper.NewContainerReportObject(put.Namespace, put.Name,
-						cut.Name, fmt.Sprintf("StartupProbe exec probe has a PeriodSeconds greater than 10 ( %d seconds)",
-							cut.StartupProbe.PeriodSeconds), true))
-				} else {
-					nonCompliantObjects = append(nonCompliantObjects,
-						testhelper.NewContainerReportObject(put.Namespace, put.Name,
-							cut.Name, fmt.Sprintf("StartupProbe exec probe has a PeriodSeconds that is not greater than 10 ( %d seconds)",
-								cut.StartupProbe.PeriodSeconds), false))
-				}
-			}
-			if cut.ReadinessProbe != nil && cut.ReadinessProbe.Exec != nil {
-				counter++
-				if CheckProbePeriodSeconds(cut.ReadinessProbe, cut, "ReadinessProbe") {
-					compliantObjects = append(compliantObjects, testhelper.NewContainerReportObject(put.Namespace, put.Name,
-						cut.Name, fmt.Sprintf("ReadinessProbe exec probe has a PeriodSeconds greater than 10 ( %d seconds)",
-							cut.ReadinessProbe.PeriodSeconds), true))
-				} else {
-					nonCompliantObjects = append(nonCompliantObjects,
-						testhelper.NewContainerReportObject(put.Namespace, put.Name,
-							cut.Name, fmt.Sprintf("ReadinessProbe exec probe has a PeriodSeconds that is not greater than 10 ( %d seconds)",
-								cut.ReadinessProbe.PeriodSeconds), false))
-				}
-			}
+			nonCompliantObjects, compliantObjects = testProbe(cut.LivenessProbe, &counter, cut, "LivenessProbe")
+
+			nonCompliant, compliant := testProbe(cut.StartupProbe, &counter, cut, "StartupProbe")
+			compliantObjects = append(compliantObjects, compliant...)
+			nonCompliantObjects = append(nonCompliantObjects, nonCompliant...)
+
+			nonCompliant, compliant = testProbe(cut.ReadinessProbe, &counter, cut, "ReadinessProbe")
+			compliantObjects = append(compliantObjects, compliant...)
+			nonCompliantObjects = append(nonCompliantObjects, nonCompliant...)
```

Comment:
```suggestion
    probesToCheck := []struct{
        probe *v1.Probe
        probeName string
    }{
         {
            probe: cut.LivenessProbe,
            probeName: "LivenessProbe",
         },
         {
           probe: cut.ReadinessProbe,
           probeName: "ReadinessProbe",
         },
         {
           probe: cut.StartupProbe,
           probeName: "StartupProbe",
         },
    }
    
    for _, probeToCheck := range probesToCheck {
      ncObjs, cObjs := testProbe(probeToCheck.probe, &counter, cut, probeToCheck.probeName)
      nonCompliantObjects = append(nonCompliantObjects, ncObjs)
      compliantObjects = append(compliantObjects, cObjs)
    }
```

Reply:
I think here empty/nil check would be added for ncObjs, cObjs before adding?

Reply:
I think `append` handles empty slices just fine.